parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -1237,12 +1185,28 @@ public void pauseIfRunning() {
     @Override
     public void resumeIfPaused() {
         if (this.state == State.paused) {
-            log.info("Transitioning paarallel consumer to state running.");
+            final String msg = "Transitioning parallel consumer to state " + running;
+            log.info(msg);
             this.state = State.running;
-            notifySomethingToDo();
+            notifySomethingToDo(new Reason(msg));
         } else {
             log.debug("Skipping transition of parallel consumer to state running. Current state is {}.", this.state);
         }
     }
 
+    /**
+     * Async tell the controller about future send results. This causes the controller to only commit results for these
+     * source records when it's potentially done blockin gon it's actor queue until the next commit time - but as these
+     * records only affect things at commit time - that's great.
+     */
+    protected void handleFutureProduceResultsAsync(List<FutureConsumeProduceResult<K, V, K, V>> results) {
+        getMyActor().tell(controller -> handleFutureProduceResultsMessage(results));
```

Comment:
Can the result straight into a concurrent queue, instead of blocking on actor work. However it’s only a performance upgrade, so actor work may be higher priority